### ISSUE TITLE
Problem with ternary operator + generics

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#337]
```
The nullness checker incorrectly reports an error for the following program. 
The non-generic case works as expected (replacing uses of 'R' with Object 
allows the test to pass).

I tested with v1.8.2.

=== Test.java ===
import javax.annotation.Nullable;

abstract class Test<R> {
  abstract R getThing(String key);
  @Nullable R m(String key) {
    return (key == null) ? null : getThing(key);
  }
}
===

$ ~/jsr308/checker-framework-1.8.2/checker/bin/javac -processor 
org.checkerframework.checker.nullness.NullnessChecker -cp 
~/jsr308/checker-framework-1.8.2/checker/dist/checker-qual.jar Test.java
...
Test.java:6: error: [conditional.type.incompatible] incompatible types in 
conditional expression.
    return (key == null) ? null : getThing(key);
                           ^
  found   : null
  required: R extends @Initialized @Nullable Object
```

Original issue reported on code.google.com by `cus...@google.com` on 16 Jun 2014 at 10:35
